modules/broker-auth/src/main/java/io/ballerina/messaging/broker/auth/authentication/authenticator/LdapAuthenticator.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+package io.ballerina.messaging.broker.auth.authentication.authenticator;
+
+import io.ballerina.messaging.broker.auth.AuthException;
+import io.ballerina.messaging.broker.auth.BrokerAuthConfiguration;
+import io.ballerina.messaging.broker.auth.authentication.AuthResult;
+import io.ballerina.messaging.broker.auth.authentication.Authenticator;
+import io.ballerina.messaging.broker.auth.authorization.UserStore;
+import io.ballerina.messaging.broker.auth.ldap.LdapAuthHandler;
+import io.ballerina.messaging.broker.common.StartupContext;
+import io.ballerina.messaging.broker.common.config.BrokerConfigProvider;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Map;
+import java.util.Objects;
+import javax.naming.NamingException;
+import javax.naming.directory.DirContext;
+
+/**
+ * Ldap authentication representation for @{@link Authenticator}.
+ */
+public class LdapAuthenticator implements Authenticator {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(LdapAuthenticator.class);
+
+    private BrokerAuthConfiguration.LdapConfiguration ldapConfiguration;
```

Comment:
Field can be converted to a local variable. Please install SonarLint if you have not already done.

Reply:
I had not installed it. Now I have.